mono/metadata/metadata.c

```diff
@@ -3363,6 +3375,26 @@ mono_metadata_get_image_set_for_aggregate_modifiers (MonoAggregateModContainer *
 	return set;
 }
 
+MonoImageSet *
+mono_metadata_merge_image_sets (MonoImageSet *set1, MonoImageSet *set2)
+{
+	MonoImage **images = g_newa (MonoImage*, set1->nimages + set2->nimages);
+	memcpy (images, set1->images, sizeof (MonoImage*) * set1->nimages);
+
+	int nimages = set1->nimages;
+	// FIXME: Quaratic
+	for (int i = 0; i < set1->nimages; ++i) {
+		int j;
+		for (j = 0; j < set2->nimages; ++j) {
```

Comment:
It seems like this merge may be wrong. If `set1` has 1 item and `set2` has 10 items, it seems like we can only ever add one item from `set2` given the add `images [nimages ++] = set2->images [i];` is in the outer loop (based on item count from `set1`).

Reply:
Yep, oops. I missed that in review, thanks for catching it.